src/image_gen/imagegen_init.cpp

```diff
@@ -56,8 +135,82 @@ std::variant<Status, ImageGenPipelineArgs> prepareImageGenPipelineArgs(const goo
         SPDLOG_LOGGER_ERROR(modelmanager_logger, "Models path does not exist: {}", args.modelsPath);
         return StatusCode::PATH_INVALID;
     }
+    bool isNPU = false;
     if (nodeOptions.has_device()) {
-        args.device = nodeOptions.device();
+        isNPU = nodeOptions.device().find("NPU") != std::string::npos;
+        // use getListOfDevices
+        auto devicesOrStatus = getListOfDevices(nodeOptions.device());
+        if (std::holds_alternative<Status>(devicesOrStatus)) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Failed to parse devices: {}", nodeOptions.device());
+            return std::get<Status>(devicesOrStatus);
+        }
+        auto devices = std::get<std::vector<std::string>>(devicesOrStatus);
+        if (devices.empty()) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "No valid devices found in: {}", nodeOptions.device());
+            return StatusCode::DEVICE_WRONG_FORMAT;
+        }
+
+        // allow only 1 or 3 devices
+        if (devices.size() != 1 && devices.size() != 3) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Invalid number of devices specified: {}. Expected 1 or 3.", devices.size());
+            return StatusCode::DEVICE_WRONG_FORMAT;
+        }
+
+        args.device = std::move(devices);
+    }
+    if (nodeOptions.has_resolution()) {
+        auto res = getListOfResolutions(nodeOptions.resolution());
+        // list all the res
+        if (std::holds_alternative<Status>(res)) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Failed to parse resolution: {}", nodeOptions.resolution());
+            return std::get<Status>(res);
+        }
+        args.staticReshapeSettings = StaticReshapeSettingsArgs(std::get<std::vector<resolution_t>>(res));
+
+        if (isNPU && args.staticReshapeSettings.value().resolution.size() > 1) {
+            SPDLOG_LOGGER_ERROR(modelmanager_logger, "NPU cannot have multiple resolutions in static settings");
+            return StatusCode::SHAPE_DYNAMIC_BUT_NPU_USED;
```

Comment:
resolution vs resolutions
In that case `args.staticReshapeSettings.value().resolution` is a single, selected resolution value or a list of possible resolutions? I don't think it's clear